src/rust/engine/src/context.rs

```diff
@@ -67,6 +67,7 @@ pub struct Core {
   pub watcher: Option<Arc<InvalidationWatcher>>,
   pub build_root: PathBuf,
   pub local_parallelism: usize,
+  pub graceful_shutdown_max_wait_time: usize,
```

Comment:
Don't love this being a part of `Core`, but didn't see an obvious other place to put it :unamused: